src/Web3Context/index.tsx

```diff
@@ -32,18 +52,55 @@ const Web3Provider: React.FC = (props) => {
   const [etherBalance, setEtherBalance] = useState<any>(null);
   const [signer, setSigner] = useState<any>();
   const [wallet, setWallet] = useState<any>({});
+  const [networkStats, setNetworkStats] = useState<any>({});
+  const [crucibles, setCrucibles] = useState(
+    [] as {
+      id: string;
+      balance: string;
+      lockedBalance: string;
+      cleanBalance?: string;
+      cleanLockedBalance?: string;
+    }[]
+  );
+  const [rewards, setRewards] = useState<any>();
   const [onboard, setOnboard] = useState<ReturnType<typeof initOnboard>>(
     null as any
   );
   const [notify, setNotify] = useState<any>(null);
-
-  const updateWallet = (wallet: any) => {
+  const updateWallet = useCallback((wallet: any) => {
```

Comment:
This function response is being cached and so it's not updating when `reloadCrucibles` is called, I think.

Reply:
I don't think this is a problem, people are not expected to change their wallet wheile using the app and they can just reload the site if they want